internal/agent/device/status/containers.go

```diff
@@ -98,6 +115,83 @@ func (c *Container) Export(ctx context.Context, status *v1alpha1.DeviceStatus) e
 	return nil
 }
 
+func (c *Container) CrioExport(ctx context.Context, status *v1alpha1.DeviceStatus) error {
+	crioExecCtx, cancel := context.WithTimeout(ctx, podmanCommandTimeout)
+	defer cancel()
+	args := []string{"ps", "-a", "--output", "json"}
+	for _, pattern := range c.matchPatterns {
+		args = append(args, "--name")
+		args = append(args, pattern)
+	}
+
+	crioOut, crioErrOut, crioExitCode := c.exec.ExecuteWithContext(crioExecCtx, crioCommand, args...)
+	if crioExitCode != 0 {
+		return fmt.Errorf("failed listing crio containers with code %d: %s", crioExitCode, crioErrOut)
+	}
+
+	var containers CrioContainerList
+	err := json.Unmarshal([]byte(crioOut), &containers)
+	if err != nil {
+		return fmt.Errorf("failed unmarshalling crio containers: %s", err)
+	}
+
+	notRunning := 0
+	runningCondition := v1alpha1.Condition{
+		Type: v1alpha1.DeviceContainersRunning,
+	}
+
+	deviceContainerStatus := make([]v1alpha1.ContainerStatus, len(containers.Containers))
+	for i, c := range containers.Containers {
+		deviceContainerStatus[i].Name = c.Metadata.Name
+		deviceContainerStatus[i].Status = c.State
+		deviceContainerStatus[i].Image = c.Image
+		deviceContainerStatus[i].Id = c.Id
+
+		if c.State != crioContainerRunning {
+			notRunning++
+		}
+	}
+
+	if notRunning == 0 {
+		runningCondition.Status = v1alpha1.ConditionStatusTrue
+		runningCondition.Reason = util.StrToPtr("Running")
+	} else {
+		runningCondition.Status = v1alpha1.ConditionStatusFalse
+		runningCondition.Reason = util.StrToPtr("NotRunning")
+		containerStr := "container"
+		if notRunning > 1 {
+			containerStr = "containers"
+		}
+		runningCondition.Message = util.StrToPtr(fmt.Sprintf("%d %s not running", notRunning, containerStr))
+	}
+
+	if status.Containers == nil {
+		status.Containers = &deviceContainerStatus
+	} else {
+		*status.Containers = append(*status.Containers, deviceContainerStatus...)
+	}
+
+	return nil
+}
+
+func (c *Container) Export(ctx context.Context, status *v1alpha1.DeviceStatus) error {
+	if _, err := exec.LookPath("podman"); err == nil {
+		err := c.PodmanExport(ctx, status)
+		if err != nil {
+			return fmt.Errorf("failed exporting podman status: %w", err)
+		}
+	}
+
+	if _, err := exec.LookPath("crictl"); err == nil {
+		err := c.CrioExport(ctx, status)
+		if err != nil {
+			return fmt.Errorf("failed exporting crio status: %w", err)
```

Comment:
is this an error or should we warn? this would mean we require podman and crictl to report status.